dlt/extract/incremental/transform.py

```diff
@@ -298,13 +319,13 @@ def __call__(
             tbl = tbl.filter(end_compare(tbl[cursor_path], end_value_scalar))
```

Comment:
currently IMO arrow is ignoring nulls when doing max function: https://arrow.apache.org/docs/python/generated/pyarrow.compute.ScalarAggregateOptions.html#pyarrow-compute-scalaraggregateoptions

and I expect they are sometimes left in the table (initial load when no filters are applied) and sometimes removed (if end value is set or initial value is set please test it)

this is indeed a problem - even in the current code. what I'd do
if the cursor column is nullable: I'd filter out all the rows with cursor null. and raise if we do not allow for that. if we include nulls I'd add them back to table at the end of processing. otherwise we do not need to do anything

Reply:
That's a pragmatic idea, I implemented that approach. Please let me know what you think.